app/api/routes.py

```diff
@@ -34,11 +36,17 @@ def resources():
 @requires_body
 @authenticate
 def post_resources():
-    validation_errors = validate_resource(request)
+    json = request.get_json()
+
+    if not isinstance(json, list):
+        return wrong_type("list of resources objects")
```

Comment:
Does this error message get sent to the client? If so, I'd prefer it say something like: 
```"request body must contain valid JSON"```.

Reply:
Great question! I'll need to add that to my examples above. The `wrong type` function you see there creates a more human-readable string. So you'll see the object be something like 
```json
{
  "apiVersion": "1.0",
  "data": null,
  "error": 1,
  "errors": [
    {
      "bad-request": {
        "message": "This endpoint accepts a list of resources"
      }
    }
  ],
  "status": "Bad Request",
  "status_code": 400
}
```

Reply:
I'm mobile right now, but I'll edit the above comment with the exact response when I get back to my computer

Reply:
I updated the above comment. This is what it looks like when you send an array to the PUT endpoint:
```json
{
  "apiVersion": "1.0",
  "data": null,
  "error": 1,
  "errors": {
    "invalid-type": {
      "message": "This endpoint accepts a resource object"
    }
  },
  "status": "Unprocessable Entity",
  "status_code": 422
}
```

Reply:
Yeah, I'd change the message to what I mentioned `resource object` doesn't mean anything to a client consumer.

Reply:
The purpose of the message is not to indicate that it's invalid JSON though, so I don't think your suggestion is quite right. The reason you would get this error is if you sent valid JSON that doesn't conform to the schema expected by the API.

Do you think I should say that? `Request body must conform to required schema` and you have to look in the docs to find out what that schema is?

Reply:
> The purpose of the message is not to indicate that it's invalid JSON though

```python
if not isinstance(json, list):
    return wrong_type("list of resources objects")
```

?

Reply:
Yeah, so that's checking if it's a list type, which is the python equivalent of a JSON array. Basically, it will reach this code and pass this check if you send a `{}` instead of a `[]`. You need to send an array of resource objects, rather than a single object.

Reply:
If you send invalid json, this line will catch that and return an error telling you that you sent invalid JSON:
```python
json = request.get_json()
```

Reply:
Sure, so I'd send a message that expresses all of this logic you just explained. Something like: `Expected an array of resource objects, but received an object`

Reply:
Ok, I can make that change. Thanks!

Reply:
Ok now it looks like this in the response:
```json
{
  "apiVersion": "1.0",
  "data": null,
  "error": 1,
  "errors": {
    "invalid-type": {
      "message": "Expected list of resources objects, but found object"
    }
  },
  "status": "Unprocessable Entity",
  "status_code": 422
}
```

It can handle if you pass an object, an integer, a number, an array, a boolean or a string. I'm not aware of any other JSON types, so anything else you pass will return an error like so:
```json
{
  "apiVersion": "1.0",
  "data": null,
  "error": 1,
  "errors": [
    {
      "missing-body": {
        "message": "You must provide a valid JSON body to use this endpoint"
      }
    }
  ],
  "status": "Unprocessable Entity",
  "status_code": 422
}
```